File: plugins/Minecraft/fabric/chatclef/result/goto/goto_command_binding.py
#20260913_kpopmodder: Freeze one request, Task, and original destination binding.
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class GotoCommandBinding:
    request_id: str
    command_message_id: str
    session_id: str
    server_connection_generation: int
    java_socket_generation: int
    task_owner: str
    task_identity: str
    operation_id: str
    request_shape: str
    target_x: int
    target_y: int
    target_z: int
    requested_dimension: str | None
    world_dimension: str

File: plugins/Minecraft/runtime/chatclef_fabric_1.20.1/src/main/java/lavi/minecraft/fabric/chatclef/bridge/command/result/gotoresult/GotoCommandBinding.java
//#if MC == 12001
package lavi.minecraft.fabric.chatclef.bridge.command.result.gotoresult;

import lavi.minecraft.task.movement.gotoresult.model.GotoTargetSnapshot;

//20260913_kpopmodder: Independently bind one admitted task to its request and both connection generations.
public record GotoCommandBinding(
        String requestId, String commandMessageId, String sessionId, long serverConnectionGeneration,
        long javaSocketGeneration, String taskOwner, String taskIdentity, String operationId,
        GotoTargetSnapshot target
) { }
//#endif
